Include/pymacro.h

```diff
@@ -56,13 +63,44 @@
    #define foo_to_char(foo)  \
        ((char *)(foo)        \
         + Py_BUILD_ASSERT_EXPR(offsetof(struct foo, string) == 0))
+   Originaly written by Rusty Russell, public domain, http://ccodearchive.net/
+   Modified to prohibit variable-length arrays(VLA). */
+#if defined(__cplusplus)
+    template<typename T>
+    struct _Py_BUILD_ASSERT_EXPR_prohibit_vla {
+        static_assert(sizeof(T) == 1,
+            "Py_BUILD_ASSERT_EXPR can only be used with constant "
+            "expression of value true");
+    };
+#  define Py_BUILD_ASSERT_EXPR(cond) \
+        (!sizeof(_Py_BUILD_ASSERT_EXPR_prohibit_vla<char[1 - 2 * !(cond)]>))
+#elif defined(_MSC_VER)
+#  define Py_BUILD_ASSERT_EXPR(cond) \
+        (!sizeof( \
+            __pragma(warning(push)) \
+            __pragma(warning(suppress: 4116)) \
+            enum { \
+                Py_CONCAT(_Py_BUILD_ASSERT_EXPR_prohibit_vla_,__LINE__) = \
+                    sizeof(char[1 - 2 * !(cond)]) \
+            } \
+            __pragma(warning(pop)) \
+        ))
+#else
+#  define Py_BUILD_ASSERT_EXPR(cond) \
+        (!sizeof( \
+            enum { \
+                Py_CONCAT(_Py_BUILD_ASSERT_EXPR_prohibit_vla_,__LINE__) = \
+                    sizeof(char[1 - 2 * !(cond)]) \
+            } \
+        ))
+#endif
 
-   Written by Rusty Russell, public domain, http://ccodearchive.net/ */
-#define Py_BUILD_ASSERT_EXPR(cond) \
-    (sizeof(char [1 - 2*!(cond)]) - 1)
-
-#define Py_BUILD_ASSERT(cond)  do {         \
-        (void)Py_BUILD_ASSERT_EXPR(cond);   \
+/* Deprecated, use static_assert instead. */
+#define Py_BUILD_ASSERT(cond)  do { \
+        Py_DEPRECATED(3.13) \
+        int _Py_BUILD_ASSERT_is_deprecated_use_static_assert_instead = 0; \
+        _Py_BUILD_ASSERT_is_deprecated_use_static_assert_instead; \
```

Comment:
This unused value is intentional in case a compile doesn't support deprecating(`Py_DEPRECATED` expands to nothing) but does support warning unused value, so users can see that it is deprecated.

Should I prepend `(void)` to the variable to make GithubActions happy?

Reply:
Resolved by surrounding the test in `Modules/_testcapimodule.c` with push-disable-pop specific warning options.
The intentional unused value is preserved.

Reply:
hey men